src/property-filter/filter-options.ts

```diff
@@ -0,0 +1,36 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { AutosuggestProps } from '../autosuggest/interfaces';
+import { OptionDefinition } from '../internal/components/option/interfaces';
+
+export function filterOptions(
+  options: readonly (AutosuggestProps.Option | AutosuggestProps.OptionGroup)[],
+  searchText = ''
+): readonly (AutosuggestProps.Option | AutosuggestProps.OptionGroup)[] {
+  if (!searchText) {
+    return options;
+  }
+
+  searchText = searchText.toLowerCase();
```

Comment:
Hypothetically, if we were to change the search algorithm using capitalized search Text, we would need to change it in `filterOptions` and `matchSingleOption`. Let' move the `.toLowerCase()` to one function. Also, let's keep it functional and not modify function parameters.

Reply:
Thanks, fixed